cgi-bin/DW/BlobStore.pm

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/perl
+#
+# DW::BlobStore
+#
+# Meta storage API for storing arbitrary blobs of content by key.
+#
+# Authors:
+#     Mark Smith <mark@dreamwidth.org>
+#
+# Copyright (c) 2016 by Dreamwidth Studios, LLC.
+#
+# This program is free software; you may redistribute it and/or modify it under
+# the same terms as Perl itself.  For a copy of the license, please reference
+# 'perldoc perlartistic' or 'perldoc perlgpl'.
+#
+
+package DW::BlobStore;
+
+use strict;
+use v5.10;
+use Log::Log4perl;
+my $log = Log::Log4perl->get_logger( __PACKAGE__ );
+
+use DW::BlobStore::LocalDisk;
+use DW::BlobStore::MogileFS;
+use DW::BlobStore::S3;
```

Comment:
You can use `LJ::ModuleLoader->require_subclasses('DW::BlobStore')` instead of listing each submodule separately. Besides being easier to read, it lets any future submodules be loaded automatically, without having to add them here.